File: leetcode/88_MergeSortedArray/lc88.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if (n == 0) return;
        while (m >= 0 || n > 0) {
            if (m == 0) {
                while (n) {
                    nums1[n-1] = nums2[n-1];
                    n--;
                }
            }
            if (n == 0) break; // if not break, m > 0 && n > 0, which means compare can be done, put last ones to array first
            if (nums1[m-1] <= nums2[n-1]) {
                nums1[m+n-1] = nums2[n-1];
                n--;
            } else {
                nums1[m+n-1] = nums1[m-1];
                m--;
            }
        }
    }
};

File: leetcode/88_MergeSortedArray/lc88.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        nums1[m:] = nums2
        nums1.sort()
        #if n == 0:
        #   return
        #len_nums1, len_nums2 = len(nums1), len(nums2)
        #while m > 0:
        #    if nums1[len_nums1-m-n] <= nums2[0]:
        #       m -= 1
        #    else:
        #       tmp = nums1[len_nums1-m-n]
        #       nums1[len_nums1-m-n] = nums2[0]
        #      nums2[0] = tmp
        #     nums2 = sorted(nums2)
        #     m -= 1
        #for i in range(len_nums1 - len_nums2, len_nums1):
        #    nums1[i] = nums2[len_nums2-n]
        #    n -= 1
